### PR TITLE
For each

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: sylt-common/src/lib.rs [hunk 1]
```diff
@@ -8,19 +8,40 @@ pub mod ty;
...
 pub use prog::Prog;
+use rc::Rc;
 pub use ty::Type;
 pub use upvalue::UpValue;
 pub use value::{MatchableValue, Value};
 
 /// A linkable external function. Created either manually or using
 /// [sylt_macro::extern_function].
-pub type RustFunction = fn(&[Value], RuntimeContext) -> Result<Value, error::RuntimeError>;
+pub type RustFunction = fn(RuntimeContext) -> Result<Value, error::RuntimeError>;
```
Comment (FredTheDino) [inline sylt-common/src/lib.rs:26]: Reminder to RuntimeError -> Error

File: sylt-machine/src/vm.rs [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
 use std::cell::RefCell;
 use std::collections::{hash_map::Entry, HashMap, HashSet};
-use std::fmt::Debug;
 use sylt_common::error::{Error, RuntimeError, RuntimePhase};
 use sylt_common::rc::Rc;
-use sylt_common::{
-    Blob, Block, BlockLinkState, Op, Prog, RuntimeContext, RustFunction, Type, UpValue,
-    Value,
-};
+use sylt_common::{Blob, Block, BlockLinkState, Frame, Machine, Op, OpResult, Prog, RuntimeContext, RustFunction, Type, UpValue, Value};
```
Comment (FredTheDino) [inline sylt-machine/src/vm.rs:7]: Long?

File: sylt-machine/src/vm.rs [hunk 3]
```diff
@@ -199,6 +178,105 @@ impl VM {
...
+
+    fn blobs(&self) -> &[Blob] {
+        &self.blobs
+    }
+
+    fn eval_call(&mut self, callable: Value, args: &[&Value]) -> Result<Value, Error> {
+        self.push(callable);
+        let num_args = args.len();
+        args.iter().for_each(|value| self.push(Value::clone(value)));
+        let ip = self.frame().ip;
```
Comment (FredTheDino) [inline sylt-machine/src/vm.rs:263]: Maybe a comment about this?

File: sylt-std/src/sylt.rs [hunk 4]
```diff
@@ -17,15 +18,70 @@ pub fn dbg<'t>(values: &[Value], _ctx: RuntimeContext<'t>) -> Result<Value, Runt
...
+        }
+        _ => {}
+    }
+    return Err(RuntimeError::ExternTypeMismatch(
+        "call".to_string(),
+        values.iter().map(Type::from).collect(),
+    ));
+}
+*/
+
```
Comment (FredTheDino) [inline sylt-std/src/sylt.rs]:
```suggestion
```

File: sylt-typechecker/src/typechecker.rs [hunk 5]
```diff
@@ -746,12 +786,15 @@ impl VM {
                             Type::Union(returns)
                         }
                     }
-                    _ => match call_callable(callable) {
-                        Err(e) => {
-                            err = Some(e);
-                            Type::Void
+                    _ => {
+                        let callable = callable.clone();
+                        match self.call_callable(callable, new_base) {
```
Comment (FredTheDino) [inline sylt-typechecker/src/typechecker.rs:791]: Return an error from `call_callable` so we can raise it directly?

**Review by FredTheDino — APPROVED** [review]

File: sylt-common/src/lib.rs (hunk 1 above)
Comment (sornas) [reply sylt-common/src/lib.rs:26]: #284

**Pull request merged**